### PR TITLE
Make rostest dependency conditional, depending on CATKIN_ENABLE_TESTING

### LINKED ISSUE #58: Mandatory dependency on rostest
Hi,

I have noticed that the dependency on rostest is always there. In other packages ([TF example](https://github.com/ros/geometry/blob/indigo-devel/tf/CMakeLists.txt#L51)) rostest is only find-packaged if `CATKIN_ENABLE_TESTING` is true. [Reference issue](https://github.com/ros/rosdistro/issues/3010)

I wonder whether there is a reason for not doing the same in this package.

### PR DESCRIPTION
**Pull request by jvgomez**
Closes #58

### CODE REVIEW

**Review by tfoote — APPROVED** [review]
lgtm

File: CMakeLists.txt [hunk 1]
```diff
@@ -33,6 +33,8 @@ target_link_libraries(state_publisher ${PROJECT_NAME}_solver ${orocos_kdl_LIBRAR
 
 if (CATKIN_ENABLE_TESTING)
 
+  find_package(rostest)
```
Comment (wjwwood) [inline CMakeLists.txt]: Please use `REQUIRED`.

**Review by wjwwood — CHANGES_REQUESTED** [review]
One small change, otherwise, lgtm.

**Comment by jvgomez** [comment]
@wjwwood I have done the requested changes.

**Review by wjwwood — APPROVED** [review]

**Pull request merged**